src/overlays/actors/ovl_Bg_Last_Bwall/z_bg_last_bwall.c

```diff
@@ -28,57 +49,217 @@ const ActorInit Bg_Last_Bwall_InitVars = {
+    DynaPolyActor_LoadMesh(play, &this->dyna, D_80C18A48[this->unk238].colHeader);
+    Collider_InitTris(play, &this->colliderTris);
+    if (Flags_GetSwitch(play, BGLASTBWALL_GET_SWITCHFLAGS(&this->dyna.actor))) {
+        Actor_MarkForDeath(&this->dyna.actor);
+    } else if (!Collider_SetTris(play, &this->colliderTris, &this->dyna.actor, &sTrisInit,
+                                 &this->colliderTrisElement)) {
+        Actor_MarkForDeath(&this->dyna.actor);
+    } else {
+        BgLastBwall_InitCollider(&sTrisInit, &this->dyna.actor.world.pos, &this->dyna.actor.shape.rot,
+                                 &this->colliderTris, D_80C18AC0[this->unk238]);
+        SubS_FillCutscenesList(&this->dyna.actor, this->cutscenes, ARRAY_COUNT(this->cutscenes));
+        func_80C187E4(this);
+    }
+}
 
-extern ColliderTrisElementInit D_80C189C0[2];
-extern ColliderTrisInit D_80C18A38;
-extern InitChainEntry D_80C18AC8[];
+void BgLastBwall_Destroy(Actor* thisx, PlayState* play) {
+    BgLastBwall* this = THIS;
 
-extern UNK_TYPE D_06000098;
+    DynaPoly_DeleteBgActor(play, &play->colCtx.dyna, this->dyna.bgId);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Last_Bwall/func_80C18240.s")
+void func_80C184EC(BgLastBwall* this, PlayState* play) {
+    f32 randVar;
+    Vec3f spD0;
+    Vec3f position;
+    s32 var_v0;
+    Vec3f spB4;
+    s32 i;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Last_Bwall/BgLastBwall_Init.s")
+    Matrix_RotateYS(this->dyna.actor.shape.rot.y, MTXMODE_NEW);
+    Matrix_RotateXS(this->dyna.actor.shape.rot.y, MTXMODE_APPLY);
+    Matrix_RotateZS(this->dyna.actor.shape.rot.y, MTXMODE_APPLY);
+    for (i = 0; i < 30; i++) {
+        switch (this->unk238) {
+            case 0:
+                spD0.x = 280.0f;
+                spD0.y = Rand_Centered() * 160.0f + 320.0f;
+                spD0.z = Rand_Centered() * 160.0f;
+                Matrix_MultVec3f(&spD0, &position);
+                Math_Vec3f_Sum(&this->dyna.actor.world.pos, &position, &position);
+                break;
+            case 1:
+                spD0.x = Rand_Centered() * 160.0f;
+                spD0.y = 400.0f;
+                spD0.z = Rand_Centered() * 160.0f;
+                Matrix_MultVec3f(&spD0, &position);
+                Math_Vec3f_Sum(&this->dyna.actor.world.pos, &position, &position);
+                break;
+        }
+        func_800BBFB0(play, &position, 50.0f, 2, Rand_ZeroOne() * 120.0f + 20.0f, Rand_ZeroOne() * 240.0f + 20.0f, 0);
+        spB4.x = Rand_ZeroOne() * 2.5f;
+        spB4.y = Rand_ZeroOne() * 2.5f + 1.0f;
+        spB4.z = Rand_ZeroOne() * 2.5f;
+        randVar = Rand_ZeroOne();
+        if (randVar < 0.2f) {
+            var_v0 = 0x60;
+        } else if (randVar < 0.6f) {
+            var_v0 = 0x40;
+        } else {
+            var_v0 = 0x20;
+        }
+        EffectSsKakera_Spawn(play, &position, &spB4, &position, -260, var_v0, 20, 0, 0, 10, 0, 0, 50, -1,
```

Comment:
Usually we call `position` `pos` or in this case `effectPos` possibly. But here it's also the acceleration, so maybe make it `effectPosAndAccel`? spB4 is `velocity`/`effectVelocity`